src/client/content_cao.cpp

```diff
@@ -506,6 +511,10 @@ void GenericCAO::removeFromScene(bool permanent)
 		m_spritenode->remove();
 		m_spritenode->drop();
 		m_spritenode = nullptr;
+	} else if (m_textspritenode) {
+		m_textspritenode->remove();
+		m_textspritenode->drop();
+		m_textspritenode = NULL;
```

Comment:
If I'm not mistaken, shouldn't this be `delete m_textspritenode` instead?

Reply:
No, the pointer is owned by Irrlicht.

Reply:
Ah ok, thanks